src/components/map/mapbox/layers/HeatmapLayer.tsx

```diff
@@ -142,7 +142,10 @@ const HeatmapLayer: FC<HeatmapLayerProps> = ({
     // to update the state and then this effect can be call again when map loaded.
     const createLayers = () => {
       const dataSource = findSuitableVisiblePoint(
-        createCenterOfMassDataSource(undefined),
+        {
+          type: "FeatureCollection",
+          features: new Array<Feature<Point>>(),
+        },
```

Comment:
`features`

Reply:
i prefer using 
```
       {
          type: "FeatureCollection",
          features: new Array<Feature<Point>>(),
        },
```
rather than `features` because it is the same as the return values of previous code (always passing in `undefined`)
```
createCenterOfMassDataSource(undefined),
```

Reply:
yes I understand you using a empty FeatureCollection here. but in destructuring props you give a default empty FeatureCollection<Point> here `features = generateFeatureCollectionFrom(undefined)` if feature is undefined, and 
```
export interface LayersProps {
  // Tile layer should added to map
  // collections?: Array<OGCCollection>;
  features?: FeatureCollection<Point>;
  // Event fired when user click on the point layer
  onDatasetSelected?: (uuids: Array<string>) => void;
  // dataset that user selected from result list or map
  selectedUuids?: string[];
  showFullMap?: boolean;
  tabNavigation?: (uuid: string, tab: string, section?: string) => void;
}
```
also ensure the passed in feature (if has) is FeatureCollection<Point>, so I think no need to write an empty FeatureCollection here can use `features` directly

in previous code, it didn't give default empty initial value in props destructure, therefore using `createCenterOfMassDataSource(undefined)` here to make sure there will be an empty FeatureCollection<Point>. so you can also use ` generateFeatureCollectionFrom(undefined)` for getting the empty FeatureCollection<Point> since you have already used it in previous code.